core/dbt/include/global_project/macros/materializations/models/table/columns_spec_ddl.sql

```diff
@@ -37,13 +37,34 @@
   {#--Obtain the column schema provided by the schema file by generating an 'empty schema' query from the model's columns. #}
   {%- set schema_file_provided_columns = get_column_schema_from_query(get_empty_schema_sql(model['columns'])) -%}
 
+  {#-- Build compiler error msg #}
   {%- set sql_file_provided_columns_formatted = format_columns(sql_file_provided_columns)  -%}
   {%- set schema_file_provided_columns_formatted = format_columns(schema_file_provided_columns)  -%}
+  {%- set compiler_error_msg = 'Please ensure the name, data_type, and number of columns in your `yml` file match the columns in your SQL file.\nSchema File Columns: ' ~ (schema_file_provided_columns_formatted|trim) ~ '\n\nSQL File Columns: ' ~ (sql_file_provided_columns_formatted|trim) ~ ' ' -%}
 
-  {%- if sql_file_provided_columns_formatted != schema_file_provided_columns_formatted -%}
-    {%- do exceptions.raise_compiler_error('Please ensure the name, data_type, order, and number of columns in your `yml` file match the columns in your SQL file.\nSchema File Columns: ' ~ (schema_file_provided_columns_formatted|trim) ~ '\n\nSQL File Columns: ' ~ (sql_file_provided_columns_formatted|trim) ~ ' ' ) %}
+
+  {%- if sql_file_provided_columns|length != schema_file_provided_columns|length -%}
+    {%- do exceptions.raise_compiler_error(compiler_error_msg) -%}
   {%- endif -%}
 
+  {%- for sql_col in sql_file_provided_columns -%}
+    {%- set yaml_col = [] -%}
```

Comment:
This won't catch when there's a column defined in the schema file but missing from sql.

Reply:
I think that either the length will be wrong or there would be the same number of columns with two different names, and both cases will cause an error message.